open_instruct/dpo_tune.py

```diff
@@ -708,14 +725,21 @@ def load_model():
                 clean_last_n_checkpoints(args.output_dir, args.keep_last_n_checkpoints)
             accelerator.wait_for_everyone()
 
-    if args.with_tracking:
-        accelerator.end_training()
-
-    # save normally at the end
     if args.output_dir is not None:
         if accelerator.is_main_process:
             tokenizer.save_pretrained(args.output_dir)
-        save_with_accelerate(accelerator, model, tokenizer, args.output_dir, args)
+        save_with_accelerate(
+            accelerator,
+            model,
+            tokenizer,
+            args.output_dir,
+            args.use_lora,
+            args.push_to_hub,
+            args.hf_repo_id,
+            args.hf_repo_revision,
+        )
+    if accelerator.is_main_process and args.launch_beaker_eval_jobs:
+        submit_beaker_eval_jobs(args.hf_repo_id, args.hf_repo_revision)
```

Comment:
does this mean that submitting the beaker eval jobs requires setting push_to_hub? It might be good to assert that upfront to avoid this erroring after a long training run.

Reply:
This is a really good catch! I have put the `launch_beaker_eval_jobs` only under the push to hub in the script and added an assert statement.